### PR TITLE
Add cancel event to webhook

### PR DESCRIPTION
**Pull request by anarosas**
Add cancel event to webhook, currently we only send notifications when the build fails

### CODE REVIEW

File: lib/travis/addons/config/notify.rb [hunk 1]
```diff
@@ -49,6 +49,13 @@ def on_failure_for?(type)
             end
           end
 
+          def on_canceled_for?(type)
```
Comment (svenfuchs) [inline lib/travis/addons/config/notify.rb]: Sorry, I haven't seen this PR before. Always better to ping me on Slack, too :)

I suppose we'd want an entry in the `DEFAULTS` hash for this, don't we? https://github.com/travis-ci/travis-hub/blob/d69767aaf17ac43fb7b4ea31f486282244943aba/lib/travis/addons/config/notify.rb#L13

Comment (svenfuchs) [inline lib/travis/addons/config/notify.rb]: Also, if we add this ... I'm wondering what about the `errored` state?

Comment (carlad) [inline lib/travis/addons/config/notify.rb]: @svenfuchs do you mean we want a webhook for errored as well?

Comment (anarosas) [inline lib/travis/addons/config/notify.rb]: @svenfuchs  Yes, I'm sorry, I worked on this over the holidays and completely forgot about it, when I got back to API and billing.
I also had that same question, do we need to separate and differentiate between cancel and errored?

**Comment by carlad** [comment]
okay I added `errored` and tested on staging: looks like it's working http://mockbin.org/bin/c5724ab6-e3dc-4b35-b839-b9b3bf2b21df/log

**Pull request merged**